### DISCUSSION TITLE
How to create a PAT with write:packages and without repo permissions?

### DISCUSSION THREAD

**Discussion by alalazo** [community/community#24930] [Actions]
<p>I am using a public repository on GitHub Free to experiment with GitHub Packages.</p>
<p>I would like to use GitHub Container Registry, as explained <a href="https://docs.github.com/en/free-pro-team@latest/packages/guides/migrating-to-github-container-registry-for-docker-images" rel="noopener nofollow ugc">here</a> but I am unable to create a Personal Access Token with <code>write:packages</code> but without <code>repo</code> permissions. As soon as I select <code>write:packages</code>, then <code>repo</code> permissions are selected and greyed out.</p>
<p>Is this expected?</p>
<p>If it matters I have enabled “Improved container support” through the “Feature Preview” menu.</p>
<p>Massimiliano</p>

**Comment by airtower-luna** [comment]
<p>That’s peculiar! I have an existing token with only <code>read:packages</code> and <code>write:packages</code> permissions, but if I try to create a new one I see the same behavior that you’re describing.</p>

**Comment by whitneyimura** [comment]
<p>:wave: Thank you for reporting this.  It looks like we have a bug on our end.  You should be able to create a PAT without the read scope for use with Container Registry.</p>
<p>In the meantime while we work on the fix, you can remove the repo scope once the PAT has been created in the edit screen.  If you go to your settings &gt; Developer Settings &gt; Select the PAT and edit, you should be good to go.</p>

**Comment by alalazo** [comment]
<p>Thank you very much. The workaround works perfectly.</p>

**Comment by dries007** [comment]
<p>I cannot un-check the repo permissions when creating or editing a PAT with only write:packages selected, so the workaround doesn’t seem to work anymore.</p>

**Comment by devinrsmith** [comment]
<p>I was able to edit a PAT a few weeks ago to deselect the <code>repo</code> scope, but this is no longer working.</p>

**Comment by jcansdale** [comment, marked as answer]
<p>A fix for this should be deployed soon.</p>
<p>For the moment, you can create a <code>write:packages</code> only PAT by navigating to the following URL:<br>
<code>https://github.com/settings/tokens/new?scopes=write:packages</code></p>
<p>I hope that helps!</p>

**Comment by risicle** [comment]
FWIW I'm still experiencing this on Firefox on a mac (but the link still works)

**Comment by qoomon** [comment]
If anyone wants to automate creating PATs you can use a simple playwrite script to do so see https://gist.github.com/qoomon/6a4075a77315d1585449283debf0fc4c
